wp-content/themes/core/components/blocks/stats/Stats_Block_Controller.php

```diff
@@ -0,0 +1,178 @@
+		}
+
+		$this->layout            = (string) $args[ self::LAYOUT ];
+		$this->content_align     = (string) $args[ self::CONTENT_ALIGN ];
+		$this->display_dividers  = (string) $args[ self::DISPLAY_DIVIDERS ];
+		$this->title             = (string) $args[ self::TITLE ];
+		$this->description       = (string) $args[ self::DESCRIPTION ];
+		$this->container_classes = (array) $args[ self::CONTAINER_CLASSES ];
+		$this->content_classes   = (array) $args[ self::CONTENT_CLASSES ];
+		$this->classes           = (array) $args[ self::CLASSES ];
+		$this->attrs             = (array) $args[ self::ATTRS ];
+		$this->stats             = (array) $args[ self::STATS ];
+		$this->init();
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function defaults(): array {
+		return [
+			self::LAYOUT            => Stats_Block::LAYOUT_STACKED,
+			self::CONTENT_ALIGN     => Stats_Block::CONTENT_ALIGN_LEFT,
+			self::DISPLAY_DIVIDERS  => Stats_Block::DISPLAY_DIVIDERS,
+			self::TITLE             => '',
+			self::DESCRIPTION       => '',
+			self::CONTAINER_CLASSES => [ 'b-stats__container', 'l-container' ],
+			self::CONTENT_CLASSES   => [ 'b-stats__content' ],
+			self::CLASSES           => [ 'c-block', 'b-stats' ],
+			self::ATTRS             => [],
+			self::STATS             => [],
+		];
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function required(): array {
+		return [];
+	}
+
+	/**
+	 * Initial setup stuff
+	 */
+	public function init() {
+		$this->classes[] = 'c-block--' . $this->layout;
+		$this->classes[] = 'c-block--' . $this->display_dividers;
+	}
+
+	/**
+	 * @param array $args
+	 *
+	 * @return array
+	 */
+	public function get_content_args(): array {
+		return [
+			'tag'     => 'header',
+			'classes' => [ 'b-stats__header' ],
+			'title'   => defer_template_part( 'components/text/text', null, [
+				'content' => $this->title,
+				'tag'     => 'h2',
+				'classes' => [ 'b-stats__title', 'h3' ],
+			] ),
+			'content' => defer_template_part( 'components/text/text', null, [
+				'content' => $this->description,
+				'classes' => [ 'b-stats__description', 't-sink', 's-sink' ],
+			] ),
+			'layout'  => Stats_Block::CONTENT_ALIGN_CENTER === $this->content_align ? Content_Block::LAYOUT_CENTER : Content_Block::LAYOUT_STACKED,
+		];
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_stats(): array {
+		$statistic_args = [];
+
+		foreach ( $this->stats as $item ) {
+			// Skip over statistic rows with no value.
+			if ( empty( $item[ Stats_Block::ROW_VALUE ] ) ) {
+				continue;
+			}
+
+			$statistic_args[] = [
+				Statistic::VALUE => defer_template_part( 'components/text/text', null, $this->get_value_args( $item ) ),
+				Statistic::LABEL => defer_template_part( 'components/text/text', null, $this->get_label_args( $item ) ),
+			];
+		}
+
+		return $statistic_args;
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function get_value_args( $item ): array {
+		return [
+			'content' => esc_html( $item[ Stats_Block::ROW_VALUE ] ),
```

Comment:
@dpellenwood you raise a good point, I could for sure see `<sup>` or `<sub>` maybe being needed here. To handle the `<sup><sub>` cases, maybe the `get_value_args()` method is modified to include the inner html needed next to the value and it then returns `defer_template_part()` instead of an array of args?

For wrapping `<span>` or another element around the text component, I don't see as many use cases.. at that point wouldn't we just modify the statistic component itself to have the wrapper? I'm not well versed in how y'all are handling components needing containers, so maybe there's an easier way to make this flexible that I'm not seeing.

Reply:
For now, you can leave this as-is, imho. I think we'll just have to figure this out as we go. Thanks, Jamie!